### PR TITLE
added restamping for tf messages

### PR DESCRIPTION
**Pull request by weidingerc**

### CODE REVIEW

File: ros2bag_tools/ros2bag_tools/filter/restamp.py [hunk 1]
```diff
@@ -72,10 +73,21 @@ def filter_msg(self, serialized_msg):
...
+                t = min(times)
         if topic in self._offset_topics:
             t = t + self._offset.nanoseconds
             if hasattr(msg, 'header') and self._modify_header:
                 header_time += self._offset
                 msg.header.stamp = header_time.to_msg()
                 data = serialize_message(msg)
+            elif isinstance(msg, TFMessage) and self._modify_header:
+                for transform in msg.transforms:
+                    if hasattr(transform, 'header'):
```
Comment (Kettenhoax) [inline ros2bag_tools/ros2bag_tools/filter/restamp.py:88]: Checking for isinstance(msg, TFMessage) is enough to check headers are present. This condition can be removed

File: ros2bag_tools/ros2bag_tools/filter/restamp.py [hunk 2]
```diff
@@ -72,10 +73,21 @@ def filter_msg(self, serialized_msg):
...
+                        header_time = Time.from_msg(transform.header.stamp)
+                        header_time += self._offset
+                        transform.header.stamp = header_time.to_msg()
+                        data = serialize_message(msg)
```
Comment (Kettenhoax) [inline ros2bag_tools/ros2bag_tools/filter/restamp.py:92]: serialize_message() should only be called per-message, not per-transform like here

**Review by Kettenhoax — CHANGES_REQUESTED** [review]
Thanks for the contribution! I've marked two open issues, otherwise we can merge this

File: ros2bag_tools/ros2bag_tools/filter/restamp.py (hunk 1 above)
Comment (weidingerc) [reply ros2bag_tools/ros2bag_tools/filter/restamp.py:88]: Should we then remove the "if" condition in Line 77 as well?

Comment (Kettenhoax) [reply ros2bag_tools/ros2bag_tools/filter/restamp.py:88]: No, we don't know the type of other messages and _just_ check for headers

Comment (Kettenhoax) [reply ros2bag_tools/ros2bag_tools/filter/restamp.py:88]: Ah, I mistakenly went by the former line number. Yes, we can remove the hasattr there as well..

**Comment by weidingerc** [comment]
I have made the necessary changes.

**Comment by Kettenhoax** [comment]
LGTM

**Pull request merged**